engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorFactoryImpl.java

```diff
@@ -512,6 +512,17 @@ private Duration getTemporalValidationTolerance(ConfigurationState configuration
 		return Duration.ZERO;
 	}
 
+	private Object getConstraintValidatorPayload(ConfigurationState configurationState, Map<String, String> properties) {
+		if ( configurationState instanceof ConfigurationImpl ) {
+			ConfigurationImpl hibernateSpecificConfig = (ConfigurationImpl) configurationState;
+			if ( hibernateSpecificConfig.getConstraintValidatorPayload() != null ) {
+				LOG.logConstraintValidatorPayload( hibernateSpecificConfig.getConstraintValidatorPayload() );
+				return hibernateSpecificConfig.getConstraintValidatorPayload();
+			}
+		}
+		return null;
```

Comment:
@gsmet question: Here I am simply returning `null` and not trying to get a result out of the `properties` (like in `getTemporalValidationTolerance(...)` above. Is that OK/enough here? As I understand looking into the `properties` is the way to handle the xml config, however does it make sense to define a payload in the xml at all? If so it would probably just be of type string (even tough the payload could be "object") Do you know what I mean? (Its late... :wink:)

Reply:
Yeah, no need to implement it in the properties. It does not make much sense.

I simplified the parameters of the method as we don't need the `properties` at all.